src/main/java/Mario/Parser.java

```diff
@@ -0,0 +1,318 @@
+                        "Please enter an integer between 0 and "
+                            + (mario.getNumTasks() + 1)));
+                return;
+            }
+            mario.markTask(number);
+        } catch (NumberFormatException e) {
+            System.out.println(ui.betweenLines(
+                    "Mamma mia! Please enter a valid integer."));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Checks if input is valid, if so calls {@link Mario#unmarkTask(int)}.
+     * @param input
+     */
+    public void handleUnmarkTask(String input) {
+        String[] n = input.split(" ");
+        if (n.length != 2) {
+            System.out.println(ui.betweenLines(
+                    "Please enter a valid number!"));
+            return;
+        }
+        try {
+            int number = Integer.parseInt(n[1]);
+            if (number > mario.getNumTasks() || number < 1) {
+                System.out.println(ui.betweenLines("Please enter an integer between 0 and "
+                        + (mario.getNumTasks() + 1)));
+                return;
+            }
+            mario.unmarkTask(number);
+        } catch (NumberFormatException e) {
+            System.out.println(
+                    ui.betweenLines(
+                            "Mamma mia! Please enter a valid integer."));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Checks if input is valid, if so calls {@link Mario#removeTask(int)}
+     * @param input
+     */
+    public void handleRemoveTask(String input) {
+        String[] n = input.split(" ");
+        if (n.length != 2) {
+            System.out.println(ui.betweenLines("Please enter a valid number!"));
+            return;
+        }
+        try {
+            int number = Integer.parseInt(n[1]);
+            if (number > mario.getNumTasks() || number < 1) {
+                System.out.println(ui.betweenLines("Please enter an integer between 0 and "
+                        + (mario.getNumTasks() + 1)));
+                return;
+            }
+            mario.removeTask(number);
+        } catch (NumberFormatException e) {
+            System.out.println(ui.betweenLines("Mamma mia! Please enter a valid integer."));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Calls {@link Mario#bye()}
+     */
+    public void handleBye() {
+        mario.bye();
+    }
+
+    /**
+     * Handles invalid input and prompts new user input.
+     */
+    public void handleInvalidInput() {
+        System.out.println(ui.betweenLines("I don't understand what that means :("));
+    }
+
+    /**
+     * Converts valid date time string to type LocalDateTime.
+     * @param dateTimeString In the format yyyy-MM-dd HHmm.
+     * @return
+     */
+    public static LocalDateTime parseDateTime(String dateTimeString) {
```

Comment:
I like that you have created a specific function to handle date-times so it is possible to change the date-time format easily :)